primo/data_parser/well_data.py

```diff
@@ -0,0 +1,1149 @@
+
+        rows: list
+            List of rows(wells) in the DataFrame
+
+        col_name : str
+            Name of the new column that contains the flag information
+        """
+        if len(rows) == 0:
+            # Nothing to flag, so return.
+            return
+
+        self.data[col_name] = 0
+        for row in rows:
+            self.data.loc[row, col_name] = 1
+
+    def add_new_columns(self):
+        """
+        Adds new columns to the DataFrame. Read -> Remove deleted rows -> Join columns
+        """
+        # NOTE: Can avoid using this method by importing all the required columns
+        # at the beginning. Just have to register unsupported columns in the
+        # WellDataColumnNames object using obj.register_new_columns() method before
+        # creating the WellData object.
+        # NOTE: Not supporting it currently, since it is not crucial. This can be
+        # supported in the future.
+        msg = (
+            "This method is not supported currently. Ensure that all the required columns "
+            "are specified in the WellDataColumnNames object. To read unsupported data, "
+            "register the corresponding column using the `register_new_columns` method."
+        )
+        raise_exception(msg, NotImplementedError)
+
+    def _check_age_availability(self):
+        """
+        Check if the age of the well is available or not. If not, then
+        this method fills the missing information.
+        """
+        # Identify wells for which age is not specified
+        col_name = self._col_names.age
+        flag, empty_cells = self.has_incomplete_data(col_name)
+        if not flag:
+            # Age is provided for all wells
+            return
+
+        # Flag the wells for which, the age is estimated/filled.
+        self.flag_wells(empty_cells, "age_flag")
+        if self.config.missing_age == "fill":
+            LOGGER.warning(
+                f"Assigning the age of the well as {self.config.fill_age} "
+                f"years, if it is missing. To change this number, pass fill_age "
+                f"argument while instantiating the WellData object."
+            )
+            self.fill_incomplete_data(col_name, self.config.fill_age)
+
+        elif self.config.missing_age == "remove":
+            self.drop_incomplete_data(col_name, "age")
+
+        elif self.config.missing_age == "estimate":
+            LOGGER.warning(
+                "Estimating the age of a well if it is missing. The estimation "
+                "approach assumes that all the wells in the input file are "
+                "arranged in chronological order of their commission. DO NOT USE "
+                "this utility if that is not the case!."
+            )
+            # age_estimation(self)
+            raise_exception(
+                "Age estimation feature is not supported currently.",
+                NotImplementedError,
+            )
+
+    def _check_depth_availability(self):
```

Comment:
This function and the prior function "_check_age_availability" seem to do the same thing, just on different columns. Is there a way to generalize into one function and then call on the respective columns?

Reply:
Consolidated both the methods into a single method.